### ISSUE TITLE
help: Keras with/without a dummy layer  intermediate layer output are different 

### ISSUE THREAD

**Issue by lxpwj** [keras-team/keras#13281]
Hi, All,

I am studying custom layer in Keras.
Based on mnist_cnn.py example, I added a dummy layer, which just pass input to output. this layer acutally does nothing.
However, the output of conv2d_1 and conv2d_2 are different.

can any one suggest why?

Here i attached my source code as well as test3.png. A switch enable_dummy_layer can turn dummy layer on/off. Thank you.

[test.zip](https://github.com/keras-team/keras/files/3573111/test.zip)

**Comment by lxpwj** [comment]
here i get some hints from https://www.researchgate.net/post/Whenever_i_run_my_neural_network_I_get_different_result 

This maybe is because your neural network is initialised with random values of gains or weights, then you'd have different starting points per each simulation during the training phase. To have always the same result, try to propose a fixed seed for the initial weights. Another option is that the neural network works with a random data division, taking randomly the order of input/target pairs to compute the error propagation phase. Try to check if your issue is coming by one of the two above possibilities or even if it is coming from both.

i will try fix those random values as well as random configurations.
